README.md

```diff
@@ -2,26 +2,68 @@
 
 ![TrueTime](truetime.png "TrueTime for Android")
 
+*⚠️ See work in progress section below. TrueTime is undergoing some changes*
+
+----------------------------------------
+
 *Make sure to check out our counterpart too: [TrueTime](https://github.com/instacart/TrueTime.swift), an NTP library for Swift.*
 
 NTP client for Android. Calculate the date and time "now" impervious to manual changes to device clock time.
 
-In certain applications it becomes important to get the real or "true" date and time. On most devices, if the clock has been changed manually, then a `new Date()` instance gives you a time impacted by local settings.
+In certain applications it becomes important to get the real or "true" date and time. On most devices, if the clock has been changed manually, then a `Date()` instance gives you a time impacted by local settings.
 
 Users may do this for a variety of reasons, like being in different timezones, trying to be punctual by setting their clocks 5 – 10 minutes early, etc. Your application or service may want a date that is unaffected by these changes and reliable as a source of truth. TrueTime gives you that.
 
 You can read more about the use case in our [blog post](https://tech.instacart.com/truetime/).
 
-In a [recent conference talk](https://vimeo.com/190922794), we explained how the full NTP implementation works with Rx. Check the [video](https://vimeo.com/190922794) and [slides](https://speakerdeck.com/kaushikgopal/learning-rx-by-example-2?slide=31) out for implementation details.
+In a [conference talk](https://vimeo.com/190922794), we explained how the full NTP implementation works (with Rx). Check the [video](https://vimeo.com/190922794#t=1466s) and [slides](https://speakerdeck.com/kaushikgopal/learning-rx-by-example-2?slide=31) out for implementation details.
+
+
+# Work in Progress 4.0
+
+With the move to Kotlin & Coroutines TrueTime 4 was a [major overhaul](https://github.com/instacart/truetime-android/pull/129). We still haven't ported some of the additional bells & whistles. This section keeps track of those features (that will come in the near future). TrueTime is completely functional without these additional features, so feel free to start using it.
+
+- [ ] Fix Jitpack import
+
+I may have busted the jitpack import. Got to fix this first.
+
+- [ ] Introduce a Cache provider
+
+* Add an `interface CacheProvider` so folks can inject in their preferred caching mechanisms
+* Provide a default cache implementation (probably using the non-android version of [DataStore](https://developer.android.com/topic/libraries/architecture/datastore#kts))
+* ? Provide example of using this with a Database like Realm
+
+- [ ] BootCompletedBroadcastReceiver sample
```

Comment:
Maybe we could mention here also that if you still need to use RxJava for Truetime, you should stick with 3.x?

Reply:
I'm thinking given coroutines is baked into most android apps out there, and TrueTime itself doesn't require you to use coroutines. You can just call `TrueTime.sync` in the `App` and be on your merry way calling `truetime.now`, i don't want to bring in the notion of Rx vs coroutines at all.